code/modules/library/random_books.dm

```diff
@@ -0,0 +1,68 @@
+/obj/item/weapon/book/manual/random/New()
+	var/static/banned_books = list(/obj/item/weapon/book/manual,/obj/item/weapon/book/manual/random,/obj/item/weapon/book/manual/nuclear,/obj/item/weapon/book/manual/wiki)
+	var/newtype = pick(typesof(/obj/item/weapon/book/manual) - banned_books)
+	new newtype(loc)
+	qdel(src)
+
+/obj/structure/bookcase/random
+	var/category = null
+	var/book_count = 2
+	anchored = 1
+	state = 2
+/obj/structure/bookcase/random/New()
+	..()
+	if(ticker && ticker.current_state >= GAME_STATE_PLAYING)
+		initialize()
+
+/obj/structure/bookcase/random/initialize()
+	if(!book_count || !isnum(book_count))
+		update_icon()
+		return
+	if(!establish_db_connection())
+		if(prob(5))
+			var/obj/item/weapon/paper/P = new(get_turf(loc))
+			P.info = "There once was a book from Nantucket<br>But the database failed us, so f*$! it.<br>I tried to be good to you<br>Now this is an I.O.U<br>If you're feeling entitled, well, stuff it!<br><br><font color='gray'>~</font>"
+		update_icon()
+		return
+	if(category && prob(25)) category = null
+	book_count += pick(-1,-1,0,1,1)

```

Comment:
There's no particular reason.  My thinking was that you generally speaking want greater variety rather than less; a bunch of bookshelves with exactly the same number of books is suspiciously regular.

This way there is a higher likelihood that two adjacent bookshelves will have counts that differ by 2 rather than by 1.  If anything there ought to be a +-2 as well, for a small chance of variance-3 and a smaller chance of variance-4.

It's the same thinking that went into dirtystation.